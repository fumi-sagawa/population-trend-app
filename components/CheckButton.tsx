import React from 'react';
import { css } from '@emotion/react';
import { cssColor, querySelector } from '../utils/cssVariables';
//custom hooks
import { useReplacePopulationData } from '../hooks/useReplacePopulationData';
import { useToggleItem } from '../hooks/useToggleItem';
//types
import { Prefecture } from '../interfaces/Prefecture';

type Props = {
  item: Prefecture;
};

export const CheckButton: React.VFC<Props> = ({ item }) => {
  //カスタムフック
  //item(単数の県データ)とselectedに応じた人口データの追加と削除
  const replacePopulationData = useReplacePopulationData(item);
  //item(単数の県データ)とstoreを照合したselectedのトグル関数
  const toggleItem = useToggleItem(item);
  //発火用関数
  const onChange = () => {
    toggleItem();
    replacePopulationData();
  };

  return (
    <>
      <div css={item.selected ? selectedCheckButton : checkButton}>
        <label>
          <input type="checkbox" checked={item.selected} onChange={onChange} />
          {item.prefName}
        </label>
      </div>
    </>
  );
};

const checkButtonBase = css`
  label {
    font-size: 16px;
    font-weight: 600;
    padding: 6px 12px;
    border-radius: 999px;
    box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
    cursor: pointer;
    //親要素で大きさを指定
    width: 100%;
    height: 100%;
    //文字中央寄せ
    display: grid;
    align-items: center;
    justify-items: center;
    @media (${querySelector.pc}) {
      font-size: 17px;
    }
  }
  input {
    display: none;
  }
`;

const checkButton = css`
  /* ベースボタンを継承 */
  ${checkButtonBase}
  label {
    color: ${cssColor.mainColor};
    background-color: ${cssColor.subColor};
  }
`;

const selectedCheckButton = css`
  /* ベースボタンを継承 */
  ${checkButtonBase}
  label {
    color: ${cssColor.subColor};
    background-color: ${cssColor.mainColor};
    box-shadow: rgba(99, 99, 99, 0.2) 10px 6px 8px 0px inset;
  }
`;
