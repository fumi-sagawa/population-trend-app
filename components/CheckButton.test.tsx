/**
 * @jest-environment jsdom
 */

import { cleanup, render } from '@testing-library/react';
import { CheckButton } from './CheckButton';
import { RecoilRoot } from 'recoil';

const samplePrefecture = {
  prefCode: 13,
  prefName: '東京都',
  selected: false,
};

describe('<CheckButton />', () => {
  afterEach(() => {
    cleanup();
  });
  //スナップショット
  test('render', () => {
    const { asFragment } = render(
      //storeを使うためRecoilで挟む
      <RecoilRoot>
        <CheckButton item={samplePrefecture} />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
