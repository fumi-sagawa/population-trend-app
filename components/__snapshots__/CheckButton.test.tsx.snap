// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckButton /> render 1`] = `
<DocumentFragment>
  <div
    class="css-ytjy93-checkButtonBase-checkButton"
  >
    <label>
      <input
        type="checkbox"
      />
      東京都
    </label>
  </div>
</DocumentFragment>
`;
